### PR TITLE
Request NWC events every 10 seconds

### PR DESCRIPTION
**Pull request by benthecarman**
Made it so we request nostr wallet connect events every 10 seconds, this makes it so we don't have to make it so blastr sends events to concurrent subscriptions. This also will help with failed invoices, we will reattempt them on the next loop.

### CODE REVIEW

File: mutiny-core/src/lib.rs [hunk 1]
```diff
@@ -184,20 +184,32 @@ impl<S: MutinyStorage> MutinyWallet<S> {
...
-                                        }
                                     }
                                 }
+                                Ok(RelayPoolNotification::Message(_url, RelayMessage::EndOfStoredEvents(_))) => {
+                                    // after we process all events, sleep for 5 seconds
+                                    // and then we will reconnect to the relay
+                                    if let Err(e) = client.disconnect().await {
+                                        log_warn!(nm.logger, "Error disconnecting from nostr relay: {e}");
+                                    }
+                                    utils::sleep(10_000).await;
```
Comment (TonyGiorgio) [inline mutiny-core/src/lib.rs:209]: Ideally this is stop compatible like this: 

```
                // wait up to 5s, checking graceful shutdown check each 1s.
                for _ in 0..5 {
                    if stop.load(Ordering::Relaxed) {
                        logging_db.stop();
                        return;
                    }
                    sleep(1_000).await;
                }
```

File: mutiny-core/src/lib.rs [hunk 2]
```diff
@@ -184,20 +184,32 @@ impl<S: MutinyStorage> MutinyWallet<S> {
...
                                             }
                                         }
-                                        Ok(None) => {} // no response
-                                        Err(e) => {
-                                            log_error!(nm.logger, "Error handling NWC request: {e}");
-                                        }
                                     }
                                 }
+                                Ok(RelayPoolNotification::Message(_url, RelayMessage::EndOfStoredEvents(_))) => {
+                                    // after we process all events, sleep for 5 seconds
```
Comment (TonyGiorgio) [inline mutiny-core/src/lib.rs:204]:
```suggestion
                                    // after we process all events, sleep for 10 seconds
```

**Review by TonyGiorgio — CHANGES_REQUESTED** [review]
Blastr will delete events after they have been sent to the client right? I think this will work to continuously request events. But if starting from an off state, then we should either: 
- add a 5 or 10 second sleep before requesting events
- introduce some wait till connected to LSP
- add some retry handling to events pulled down from the relay

**Comment by benthecarman** [comment]
> Blastr will delete events after they have been sent to the client right? 

It deletes after the client sends a response event

**Comment by TonyGiorgio** [comment]
> > Blastr will delete events after they have been sent to the client right?
> 
> It deletes after the client sends a response event

Okay that makes sense actually. I think after the minor changes above it should be good to go then.

**Comment by benthecarman** [comment]
Changed so it waits for it to have usable channels and now the 10 second sleep between runs works with stops

File: mutiny-core/src/lib.rs [hunk 3]
```diff
@@ -154,6 +154,17 @@ impl<S: MutinyStorage> MutinyWallet<S> {
                     continue;
```
Comment (TonyGiorgio) [inline mutiny-core/src/lib.rs:154]: This should be a break not a continue.

**Review by TonyGiorgio — CHANGES_REQUESTED** [review]

**Review by TonyGiorgio — APPROVED** [review]

**Pull request merged**